codex/stores/repostore.nim

```diff
@@ -256,7 +256,8 @@ method listBlocks*(
     of BlockType.Block: CodexBlocksKey
     of BlockType.Both: CodexRepoKey
 
-  without queryIter =? (await self.repoDs.query(Query.init(key))), err:
+  let query = Query.init(key, value=false)
+  without queryIter =? (await self.repoDs.query(query)), err:
```

Comment:
Timing regressions on integration tests? 

Probably better would be a separate API for "query and don't return files". Alternatively, requiring an explicit option like `Query.init(key, kind=Full)` or `Query.init(key, kind=MetaOnly)`.

Reply:
I would just put a `doAssert` in the code that checks that val is empty.

Reply:
> Timing regressions on integration tests?
> 
> Probably better would be a separate API for "query and don't return files". Alternatively, requiring an explicit option like `Query.init(key, kind=Full)` or `Query.init(key, kind=MetaOnly)`.

Yeah, perhaps value should be opt-in, instead of opt-out.

Reply:
Captured this in issue #523 
https://github.com/codex-storage/nim-codex/issues/523